src/app/components/BibPage/ElectronicResources.jsx

```diff
@@ -1,22 +1,45 @@
-import { Card, CardContent, CardHeading } from '@nypl/design-system-react-components'
+import { Card, CardContent, CardHeading, Button, Icon } from '@nypl/design-system-react-components'
 import generateERLinksList from '../../utils/electronicResources'
-import React from 'react';
+import React, { useRef, useState } from 'react';
 
 /**
  * ElectronicResources renders a list of electronic resources links, sans aeon links
  * @param {array} electronicResources - an array of electronic resources, passed in as a prop from the BibPage component
+ * @param {boolean} isTestMode - a boolean value only used while running unit tests on this component
  */
-const ElectronicResources = ({ electronicResources }) => {
+
+
+const ElectronicResources = ({ electronicResources, isTestMode = false }) => {
+  const [showMore, setShowMore] = useState(true)
+  const scrollToRef = useRef();
+
   if (!electronicResources || !electronicResources.length) {
     return null;
   }
-  const resources = generateERLinksList(electronicResources)
-  return (<Card isBordered padding="16px">
+
+  const allResources = generateERLinksList(electronicResources)
+  const threeResources = generateERLinksList(electronicResources.slice(0, 3))
```

Comment:
This isn't wrong but I think it can be updated to reduce extra rendering. Right now, `allResources` and `threeResources` can both contain a `ul` with list items. Essentially, there are two lists that are being switched based on how many items should be shown. Instead, only one list should be rendered and you can manage how many of those list items are rendered.

So let's say list A has `1, 2, 3` and list B has `1, 2, 3, 4, 5, 6`. What is happening now is that both lists will separately render "1", "2", "3". Ideally, one list will be rendered with "1", "2", and "3" and the rest of the list items will render or be removed.

Let's add `electronicResources` to a state variable:
```
const [updatedElectronicResources, setUpdatedElectronicResources] = useState(electronicResources);
```

You can update this inside the `setShowMore` before the return statement.
```
setUpdatedElectronicResources(!prev ? electronicResources.slice(0,3): electronicResources);
```

Now you can call `generateERLinksList` once. The above code isn't complete and you may need to add some tweaks. Let me know if something isn't clear.